core-web/libs/sdk/client/README.md

```diff
@@ -1,163 +1,406 @@
-});
+1. **Validating the dotCMS URL**: Ensures the URL is properly formatted and accessible
+2. **Setting up authentication**: Configures the authorization header with the provided token
+3. **Creating client instances**:
+   - Initializes the Page API client for page operations
+   - Initializes the Navigation API client for navigation structures
+   - Initializes the Content API client for content operations
+
+The client maintains a single connection to your dotCMS instance, making it efficient to reuse across your application. It's recommended to:
+
+- Create a single client instance at the application level
+- Reuse this instance throughout your components/services
+- Avoid creating new client instances for each request
+
+### Page API: `client.page.get()`
+
+#### Description:
+
+The Page API allows you to retrieve pages from your dotCMS instance along with their associated content. This method returns a structured response containing both the page asset and any additional content requested through GraphQL.
+
+#### Method Signature:
+
+```typescript
+get<T>(url: string, options?: DotCMSPageRequestParams): Promise<DotCMSComposedPageResponse<T>>
 ```
 
+#### Parameters:
 
-### Warning
+| Parameter | Type | Required | Description |
+|-----------|------|----------|-------------|
+| `url` | `string` | Yes | The path/URL of the page to retrieve (e.g., '/about-us') |
+| `options` | `DotCMSPageRequestParams` | No | Configuration options for the page request |
 
-If you are updating from a version that is lower than `0.0.1-beta.29`, be aware that the response from the `client.page.get` method changed the access to the page value from `page` to `pageAsset`.
-This change was made to avoid redundancy on access inside of `page` object.
+#### DotCMSPageRequestParams Options:
 
-#### Before
+| Option | Type | Required | Default | Description |
+|--------|------|----------|---------|-------------|
+| `siteId` | `string` | No | From client config | ID of the site you want to interact with |
+| `mode` | `string` | No | 'LIVE' | Page rendering mode: 'EDIT_MODE', 'PREVIEW_MODE', or 'LIVE' |
+| `languageId` | `string\|number` | No | '1' | Language ID for content localization |
+| `personaId` | `string` | No | — | ID of the persona for personalized content |
+| `fireRules` | `boolean\|string` | No | false | Whether to execute rules set on the page |
+| `depth` | `0\|1\|2\|3\|'0'\|'1'\|'2'\|'3'` | No | '0' | Depth of related content to retrieve via Relationship fields |
+| `publishDate` | `string` | No | — | Publication date for the requested page |
+| `variantName` | `string` | No | — | Name of the specific page variant to retrieve |
+| `graphql` | `object` | No | — | GraphQL options for extending the response |
+
+#### GraphQL Options:
+
+| Option | Type | Description |
+|--------|------|-------------|
+| `page` | `string` | GraphQL query to extend the page response |
+| `content` | `Record<string, string>` | Named GraphQL queries to fetch additional content |
```

Comment:
its `unknown` unless you type it using generics

Reply:
Done!